In [6]:
import os
for f in os.listdir('../fun/'): exec(open('../fun/'+f).read())
del f

# Load data. Change winner to binary 1/0:
load( '../out/d3-fight-level-standardize-normalize.pkl' )
y[ y == -1 ] = 0
print( X.shape )

(2305, 114)


In [10]:
# train test split.
from sklearn.model_selection import train_test_split
X_train , X_test, y_train, y_test = train_test_split(
    X, y, 
    random_state = 744,
    test_size = 0.3
)

In [11]:
# Fit model.
from sklearn.ensemble import AdaBoostClassifier as classifier
m = classifier( random_state = 1120 )
m.fit( X_train, y_train )

# accuracy against train data (in-model).
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( y_train, m.predict(X_train) ) )

              precision    recall  f1-score   support

           0       0.68      0.57      0.62       585
           1       0.78      0.85      0.81      1028

    accuracy                           0.75      1613
   macro avg       0.73      0.71      0.71      1613
weighted avg       0.74      0.75      0.74      1613



In [12]:
# accuracy against test data.
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( y_test, m.predict(X_test) ) )

              precision    recall  f1-score   support

           0       0.45      0.36      0.40       263
           1       0.65      0.73      0.69       429

    accuracy                           0.59       692
   macro avg       0.55      0.54      0.54       692
weighted avg       0.57      0.59      0.58       692



f1-score in-model is higher than on test indicating **over-fitting**. 

Test-set precision is still not great at 69% (predicting all-wins give 60%).